errdefs/helpers.go

```diff
@@ -122,6 +122,26 @@ func Forbidden(err error) error {
 	return errForbidden{err}
 }
 
+type errNotAcceptable struct{ error }
+
+func (errNotAcceptable) NotAcceptable() {}
```

Comment:
I wonder if this is necessary over using `InvalidParameter`?
How will a client respond to the two things differently?
I guess in this case it would strictly be a client misconfiguration as opposed to a user providing invalid/incorrect data.

Reading up on the usage of the error code from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/406 it also seems the recommendation is to go ahead and send an OK response anyway and leave it to the client to understand that the media type is not what they asked for.

Thinking and reading maybe my suggestion to use accept headers was not useful after all.

1. An older client will continue to get old media types
2. A new client that did not make changes for the new API (or an un-versioned client) is technically incorrect but likely won't be checking mediatype anyway and will use old behavior (calling `/containers/<id>/json`)
3. A "proper" new client will look at the mediatype returned

I guess my original thought around using accept headers was to help ensure compatibility for clients in camp 2.
I was also thinking that we would return the type of stream that was requested in the accept header in all cases, but that's not actually the case and wouldn't really work anyway (at least not well).

So... is camp 2 worth it?